src/background/services/accounts/AccountsService.ts

```diff
@@ -98,8 +100,27 @@ export class AccountsService implements OnLock, OnUnlock {
     // refresh addresses so in case the user switches to testnet mode,
     // as the BTC address needs to be updated
     this.networkService.developerModeChanged.add(this.onDeveloperModeChanged);
+
+    // TODO(@meeh0w):
+    // Remove this listener after E-upgrade activation on Fuji. It will be no longer needed.
+    this.networkService.uiActiveNetworkChanged.add(
```

Comment:
Since addresses between Mainnet, Fuji & Devnet differ in prefixes, we need to not only refresh the addresses on `isTestnetMode` changes, but also if we detect the user switching to or away from the devnet.